src/gluonnlp/data/transforms.py

```diff
@@ -987,10 +990,11 @@ class BERTTokenizer:
 
     _special_prefix = u'##'
 
-    def __init__(self, vocab, lower=True, max_input_chars_per_word=200):
+    def __init__(self, vocab, lower=True, max_input_chars_per_word=200, optimize=False):
```

Comment:
What's the use-case of `optimize=False`? Would it make sense to remove the argument and always optimize?

Reply:
If so, the LRU cache size should probably be configurable?

Reply:
Both davis and I are not sure about how to pass variable max_size to `@functools.lru_cache(maxsize=DEFAULT_CACHE_SIZE)`. Currently the way ppl configure it is by setting `nlp._constant.DEFAULT_CACHE_SIZE`. @leezu any suggestion for better ways to configure it? ^^

Reply:
@davisliang I guess we have the cython dependency already, we can have always use `optimize=True`?

Reply:
Yes we can default to optimize=True. @leezu, can you advise on passing the max cache size variable on construction?

Reply:
The current solution applies the function decorator on class declaration time, thus maxsize can't be a constructor argument. One way to avoid this issue is to apply the function decorator only in `__init__`, ie. adding the following expression
```python
self._word_to_wordpiece_optimized = functools.lru_cache(maxsize=cache_maxsize)(self._word_to_wordpiece_optimized)
```

to `__init__` and deleting line 1023 `@functools.lru_cache(maxsize=DEFAULT_CACHE_SIZE)`.

With respect to making `optimize=True` the default, why not delete the non-optimized implementation, always optimize and delete the `optimize` argument?